security/pkg/nodeagent/sds/sdsservice_test.go

```diff
@@ -492,6 +493,9 @@ func TestStreamSecretsMultiplePush(t *testing.T) {
 		t.Fatalf("failed to send push notificiation to proxy %q", conID)
 	}
 	notifyChan <- "receive secret"
+	// Wait to make sure client has called Recv().
+	time.Sleep(2 * time.Second)
```

Comment:
Any test that calls sleep are not a good idea. Our nodes will occasionally get extremely CPU loaded and things run extremely slowly, so there is no amount of time that is always safe. Is there any alternative?

Reply:
Add metrics into mock secretCache, and check the metrics instead of sleep.